### PR TITLE
Redirects when multiple slashes

### PR DESCRIPTION
**Pull request by ErimTuzcuoglu**
Closes: #4422

- [ ] Docs
- [x] Tests

Testing Strategy: New and existing tests pass. We can try slashed routes manually.

### CODE REVIEW

File: packages/remix-express/server.ts [hunk 1]
```diff
@@ -52,6 +52,9 @@ export function createRequestHandler({
     next: express.NextFunction
   ) => {
     try {
+      let cleanUrl = req.url.replace(/\/+/g, '/');
+      if(req.url !== cleanUrl) return res.redirect(cleanUrl);
```
Comment (machour) [inline packages/remix-express/server.ts:56]:
```suggestion
      if (req.url !== cleanUrl) {
        return res.redirect(cleanUrl);
      }
```

Comment (ErimTuzcuoglu) [reply packages/remix-express/server.ts]: Updated

**Comment by brophdawg11** [comment]
Is this something we need to consider for servers/runtimes other than express?

**Review by machour — CHANGES_REQUESTED** [review]
@ErimTuzcuoglu can you please fix the conflict in the contributors files? 🙏🏼 
Protip: put your name in the right spot alphabetically, conflicts will be less likely to happen

**Comment by brophdawg11** [comment]
@ErimTuzcuoglu I didn't dig too deep, but was able to reproduce this using the Architect template as well?  I'm wondering this is something that should be handled in a more central location, or at least should this logic go into each of the adapters.  I will probably defer to one of the Remix team members here with more experience in the alternative runtimes.  @jacob-ebey @mcansh?

**Comment by mcansh** [comment]
> I'm wondering this is something that should be handled in a more central location

i just tried a [github](https://github.com//mcansh//snkrs//tree//main//app) url and it resolved with the multiple slashes without redirecting - perhaps we follow suit and normalize while matching?

**Comment by ErimTuzcuoglu** [comment]
The main problem is actually on URL constructor. It doesn't accept multiple slashes as a url and throws error. So when we removed slashes it works correctly, but for redirection I think we must make different implementations on adapters. As @mcansh said, on vercel we don't need any changes. But if we need I can check also other adapters and fix them too.

image

**Comment by mcansh** [comment]
> The main problem is actually on URL constructor. It doesn't accept multiple slashes as a url and throws error. So when we removed slashes it works correctly, but for redirection I think we must make different implementations on adapters. As @mcansh said, on vercel we don't need any changes. But if we need I can check also other adapters and fix them too.
> 
> 
> 
> image
> 
> 

whats interesting is that `new URL("https://github.com///")` seems to be fine image

**Review by machour — CHANGES_REQUESTED** [review]
@ErimTuzcuoglu Please update other adapters as well and rebase this PR to latest dev 🙏🏼

**Comment by brophdawg11** [comment]
Closing in favor of https://github.com/remix-run/remix/pull/5336

**Pull request closed without merging**